src/Features/Core/Portable/PullMemberUp/PullMemberUpAnalyzer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal class PullMembersUpAnalysisBuilder
+    {
+        internal static AnalysisResult BuildAnalysisResult(
+            INamedTypeSymbol targetSymbol,
+            IEnumerable<ISymbol> members)
+        {
+            var memberResult = members.Select(member =>
+            {
+                if (targetSymbol.TypeKind == TypeKind.Interface)
+                {
+                    return new MemberAnalysisResult(
+                        member, member.DeclaredAccessibility != Accessibility.Public,
+                        member.IsStatic);
+                }
+                else
+                {
+                    var changeOriginToPublic = false;
+                    var changeOriginToNonStatic = false;
```

Comment:
do you mean "ChangeOriginal"?

Reply:
these locals don't seem to serve any purpose.  just inline but use named parameters below.